### PR TITLE
Astraphobia Trigger Option

### PR DESCRIPTION
**Pull request by decodinatorX**
Astraphobia, is the fear of thunderstorms ranging from a simple few roars of thunder to intense crashing and rumbling beneath one's feet. This option is toggleable within the options menu and the only thing it does is stop the thunder and lighting effects that Week 2 gives off for those who may be a little ill toward thunderstorms, it is automatically on as the option is called "Astraphobia Triggers (On/Off)" The option is on as it means that the effects that can trigger the attacks of one are, at the moment, on.

### CODE REVIEW

**Comment by napstaa967** [comment]
hey dude, you can turn off distractions

**Comment by decodinatorX** [comment]
> hey dude, you can turn off distractions

Right, I had a feeling somebody would say that. I do know that there is a "distractions option" and if this pull request is ignored because of it I won't throw a fit over it, I just made it because even though I dislike the thunder sounds myself doesn't mean I don't like seeing and hearing the train in the back of week 3 fly past. I am perfectly fine with you pointing it out however!

**Comment by ACardboardBox1** [comment]
I feel like you should just
Remove the code of week 3's train noises being only played with distractions on
it's the flxg.save.data.distractions shit
Just remove it and you should be good to go

**Comment by decodinatorX** [comment]
> I feel like you should just
> Remove the code of week 3's train noises being only played with distractions on
> it's the flxg.save.data.distractions shit
> Just remove it and you should be good to go

True, that would work, but I had my mindset for the players of the game who may not know too much about the inner workings of the coding scene. Like I have said, I do not fault, and will not fault anyone for overlooking this kind of feature, because yes, it is technically already included with the Distractions option. But yes, that is a viable option.

**Comment by ajredden** [comment]
I tend to agree with decodinator here. Most players don't know how to code, it would be silly to force them to turn off all distractions if they're only afraid of one of them

**Review by raf13lol — COMMENTED** [review]
I am a bit in the middle of approving this pr or requesting changes because the fact is there is a phobia of trains (Siderodromophobia) so if someone liked the lightning because of atmosphere but disliked the trains due to their phobia they would have to turn off distractions. This would disable the lightning and the trains so I would probably approve this if you added a Siderodromophobia option. This is still good, and my change request is just an idea.

**Comment by decodinatorX** [comment]
> I am a bit in the middle of approving this pr or requesting changes because the fact is there is a phobia of trains (Siderodromophobia) so if someone liked the lightning because of atmosphere but disliked the trains due to their phobia they would have to turn off distractions. This would disable the lightning and the trains so I would probably approve this if you added a Siderodromophobia option. This is still good, and my change request is just an idea.

I ended up thinking about that afterwards, and I'm thinking I should go back over the code but make an extra sub-category based off the distractions thing and edited what things you can disable rather than just putting people with Siderodromophobia in the dark like I have with this simple bit of code. I may try that instead if this PR is denied.

>>EDIT:
--I have decided that the idea above is what I am going to do.

**Comment by isakube** [comment]
ayo

**Comment by CaptainGame17** [comment]
It'd probably be a better solution just to be able to toggle which distractions you want off in a sub-menu.

**Comment by steviegt6** [comment]
Incredibly odd to cater to a single niche grouping of people while leaving everyone else in the dark regarding the implementaiton. Why not just implement toggles for most non-gameplay elements? This helps people regardless of if they have any sort of irrational fear or other forms of dislike.

File: source/PlayState.hx [hunk 1]
```diff
@@ -4503,14 +4503,17 @@ class PlayState extends MusicBeatState
 
 	function lightningStrikeShit():Void
 	{
-		FlxG.sound.play(Paths.soundRandom('thunder_', 1, 2));
-		halloweenBG.animation.play('lightning');
+		if(FlxG.save.data.astraphobia)
```
Comment (steviegt6) [inline source/PlayState.hx:4506]: A little nit-picky, but an early return here is cleaner.

**Comment by AbnormalPoof** [comment]
no stable
only master

**Comment by decodinatorX** [comment]
> Incredibly odd to cater to a single niche grouping of people while leaving everyone else in the dark regarding the implementaiton. Why not just implement toggles for most non-gameplay elements? This helps people regardless of if they have any sort of irrational fear or other forms of dislike.

Hi, yes, I actually did touch on that in an later comment posted before yours here. Saying that I planned on going back and adding a submenu where one can just toggle mostly all the different gameplay "distractions" as labeled in code and game, that are fears AND can be avoided without causing the game to be altered TOO drastically. Of course if someone has a phobia of music like Melophobia is unavoidable and frankly I am dumbfounded that there even would be someone complaining about such an unavoidable thing.

**Pull request closed without merging**